org.eclipse.xtext.purexbase/src-gen/org/eclipse/xtext/purexbase/services/PureXbaseGrammarAccess.java

```diff
@@ -401,20 +409,20 @@
 		
 		//@Override
 		//XAssignment xbase::XExpression:
-		//	{xbase::XAssignment} feature=[types::JvmIdentifiableElement|FeatureCallID] OpSingleAssign value=XAssignment |
-		//	XConditionalExpression (=> ({xbase::XBinaryOperation.leftOperand=current}
-		//	feature=[types::JvmIdentifiableElement|OpMultiAssign]) rightOperand=XAssignment)?;
+		//	{xbase :: XAssignment} feature=[types::JvmIdentifiableElement|FeatureCallID] OpSingleAssign value=XAssignment |
+		//	XConditionalExpression (=> ({xbase :: XBinaryOperation . leftOperand= current} feature=
```

Comment:
The guilty is the following code (I commented the "bad" lines):
```
	protected void format(CrossReference ref, IFormattableDocument doc) {
//		doc.prepend(regionFor(ref).keyword("["), it -> it.autowrap());
		doc.append(regionFor(ref).keyword("["), it -> it.noSpace());
		doc.prepend(regionFor(ref).keyword("]"), it -> it.noSpace());
//		doc.append(regionFor(ref).keyword("]"), it -> it.autowrap());
		doc.surround(regionFor(ref).keyword("|"), it -> it.noSpace());
		doc.format(ref.getType());
	}
```
I have run most tests locally and it seems I can remove the autowrap without breaking anything else. BUT ... the rule is a bit unfortunate ... the original looks like this:

```
@Override 
XAssignment returns xbase::XExpression :
	{xbase::XAssignment} feature=[types::JvmIdentifiableElement|FeatureCallID] OpSingleAssign value=XAssignment |
	XConditionalExpression (
		=>({xbase::XBinaryOperation.leftOperand=current} feature=[types::JvmIdentifiableElement|OpMultiAssign]) rightOperand=XAssignment
	)?;
```

With above formatting I get the following:

```
@Override
XAssignment returns xbase::XExpression:
	{xbase::XAssignment} feature=[types::JvmIdentifiableElement|FeatureCallID] OpSingleAssign value=XAssignment |
	XConditionalExpression (=>
	({xbase::XBinaryOperation.leftOperand=current} feature=[types::JvmIdentifiableElement|OpMultiAssign])
	rightOperand=XAssignment)?;
```

Not THAT nice ... happens because the open paranthesis is the first possible autowrap point ...

Reply:
I added autowrap before and after Actions (braces) ... this results in the following ... guess that is as good as we can get.

```
@Override
XAssignment returns xbase::XExpression:
	{xbase::XAssignment} feature=[types::JvmIdentifiableElement|FeatureCallID] OpSingleAssign value=XAssignment |
	XConditionalExpression (=>({xbase::XBinaryOperation.leftOperand=current}
	feature=[types::JvmIdentifiableElement|OpMultiAssign]) rightOperand=XAssignment)?;
```